### PR TITLE
JS2 error improvements

### PR DESCRIPTION
**Pull request by robertbastian**

### CODE REVIEW

File: tool/src/js/mod.rs [hunk 1]
```diff
@@ -29,6 +29,16 @@ pub fn gen_bindings(
...
+    outs.get_mut("diplomat-runtime.mjs")
+        .unwrap()
+        .write_str(concat!(
+            "export class FFIError extends Error {\n",
+            "	constructor(error_value) {\n",
+            "	super('Error over FFI');\n",
+            "	this.error_value = error_value; // (2)\n",
+            "	}\n",
+            "}\n",
+        ))?;
```
Comment (ambiguousname) [inline tool/src/js/mod.rs:41]: Question: Do we still want to use FFIError? I think replacing `throw diplomatRuntime.FFIError` with `throw Error` was a good move, but I can't really see a use case for keeping FFIError around.

**Review by ambiguousname — COMMENTED** [review]
I only have a question about the presence of FFIError, even though it's no longer used.

Everything else looks good to me.

File: tool/src/js/mod.rs (hunk 1 above)
Comment (robertbastian) [reply tool/src/js/mod.rs:41]: I don't want to make this change in JS1, so adding this back to the runtime here.

Comment (ambiguousname) [reply tool/src/js/mod.rs:41]: Ahhh. Okay! Everything looks good to me then.

**Review by ambiguousname — APPROVED** [review]

File: tool/src/js/mod.rs [hunk 2]
```diff
@@ -29,6 +29,16 @@ pub fn gen_bindings(
     outs.entry("diplomat-runtime.mjs".to_string())
         .or_default()
         .write_str(include_str!("../../templates/js2/runtime.mjs"))?;
+    outs.get_mut("diplomat-runtime.mjs")
+        .unwrap()
+        .write_str(concat!(
+            "export class FFIError extends Error {\n",
```
Comment (Manishearth) [inline tool/src/js/mod.rs:35]: We eventually want to mark types as errors instead, yes? https://github.com/rust-diplomat/diplomat/issues/446

**Review by Manishearth — APPROVED** [review]

File: tool/src/js/mod.rs (hunk 2 above)
Comment (robertbastian) [reply tool/src/js/mod.rs:35]: I'm not sure, the `cause` solution seems nice enough. TS won't be able to strongly type the errors anyway.

**Pull request merged**